test/common/config.go

```diff
@@ -16,32 +16,40 @@ limitations under the License.
 
 package common
 
-import "github.com/knative/eventing/pkg/reconciler/namespace/resources"
+import (
+	"github.com/knative/eventing/test/base"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DefaultClusterChannelProvisioner is the default ClusterChannelProvisioner we will run tests against.
+const DefaultClusterChannelProvisioner = base.InMemoryProvisioner
 
 // ValidProvisionersMap saves the provisioner-features mapping.
 // Each pair means the provisioner support the list of features.
-var ValidProvisionersMap = map[string][]Feature{
-	InMemoryProvisioner:  {FeatureBasic},
-	GCPPubSubProvisioner: {FeatureBasic, FeatureRedelivery, FeaturePersistence},
-	KafkaProvisioner:     {FeatureBasic, FeatureRedelivery, FeaturePersistence},
-	NatssProvisioner:     {FeatureBasic, FeatureRedelivery, FeaturePersistence},
+var ValidProvisionersMap = map[string]ChannelConfig{
+	base.InMemoryProvisioner: ChannelConfig{
+		Features: []Feature{FeatureBasic},
```

Comment:
Although this is for kafka, I think you can add CRDSupported for the others as well. Unless that triggers the tests, which might conflict with Akash's stuff.

Reply:
It will trigger the test for CRD channels. So I will add them in the following PRs one by one.